packages/arb-validator/checkpointing/indexedCheckpointer.go

```diff
@@ -353,60 +294,41 @@ func (cp *IndexedCheckpointer) cleanupDaemon() {
 	defer ticker.Stop()
 	for {
 		<-ticker.C
-		func() {
-			cp.Lock()
+		cp.Lock()
+		cp.cleanup()
+		cp.Unlock()
+	}
+}
+
+func (cp *IndexedCheckpointer) cleanup() {
+	bounds, err := cp.getHeightBounds()
+	if err != nil {
+		return
+	}
+	height := common.NewTimeBlocks(new(big.Int).Sub(bounds.lo.AsInt(), big.NewInt(1)))
+	heightLimit := common.NewTimeBlocks(new(big.Int).Sub(bounds.hi.AsInt(), cp.maxReorgHeight))
+	var prevIds [][]byte
+	for height.Cmp(heightLimit) < 0 {
+		dbKeys := cp.db.GetKeysWithPrefix(makeContentsHeightPrefix(height))
+		if len(dbKeys) > 0 {
+			for _, id := range prevIds {
+				_ = cp.deleteCheckpointForKey(id)
```

Comment:
That isn't valid any longer. instead I just have the whole function operate with a lock on. It didn't seem like anything was gained by the fine grain lock control in the current implementation. I removed the locking from `deleteCheckpointForKey`

Reply:
I think the lock needs to be released more often here.  Holding the lock the whole time will block the rollup manager while a cleanup is running.  (The rollup manager calls AsyncWriteCheckpoint for every new on-chain block, and that blocks until the lock is available.). 

The cleanup daemon will typically be deleting order of 10 checkpoints, so it could run for a significant time.